libpod/container_api.go

```diff
@@ -581,10 +581,6 @@ func (c *Container) Wait(ctx context.Context) (int32, error) {
 // WaitForExit blocks until the container exits and returns its exit code. The
 // argument is the interval at which checks the container's status.
 func (c *Container) WaitForExit(ctx context.Context, pollInterval time.Duration) (int32, error) {
-	if !c.valid {
-		return -1, define.ErrCtrRemoved
-	}
```

Comment:
@mheon PTAL

Not sure how I feel about this, the container struct is invalid at this point taking the lock, taking the lock later etc... is no longer correct.

Reply:
You are correct. The locking model is explicitly built to bail the moment a container struct is invalid. Another container may already hold this lock number, which can lead to us deadlocking ourselves if we do anything after this point that takes a container lock. I don't see anything right now that does that, but that doesn't mean it won't be added in the future.

Reply:
I saw that the code in this case would just end up in `c.runtime.state.GetContainerExitCode()` and retrieve the status from there; but sure let's make this safer.  I'll just call `c.runtime.state.GetContainerExitCode()` if `!c.valid`, would that work for you?

Reply:
sounds good to me at least